### PR TITLE
Admin endpoint: return more resources

### PR DESCRIPTION
**Pull request by jenshu**
# Description

Part 2 of updates to admin server input snapshot endpoint:
- Add Settings (passed from bootstrap opts)
- Add GatewayParameters, ListenerOptions, HttpListenerOptions when kubeGateway is enabled
- When kubeGateway is enabled, return all RouteOptions, VirtualHostOptions, AuthConfigs, RateLimitConfigs from the cluster rather than only the ones from the edge input snapshot (since the kube gateway controller watches all resources on the cluster)

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: projects/gloo/pkg/servers/iosnapshot/history.go [hunk 1]
```diff
@@ -83,19 +90,44 @@ func (h *historyImpl) GetInputSnapshot(ctx context.Context) ([]byte, error) {
...
+	}
+
+	// get the resources from the edge api snapshot
 	resources, err := snapshotToKubeResources(snap)
 	if err != nil {
 		return nil, err
 	}
 
+	// get settings, which is not part of the api snapshot
+	if h.settings != nil {
```
Comment (jbohanon) [inline projects/gloo/pkg/servers/iosnapshot/history.go:114]: Should it be considered an error if we have nil settings? Is there a valid state of the system where we would have passed in a nil to the `NewHistory` constructor?

File: projects/gloo/pkg/servers/iosnapshot/history_test.go [hunk 2]
```diff
@@ -135,16 +351,218 @@ var _ = Describe("History", func() {
...
+								},
+							},
+						},
+					},
+				})
+
+				// k8s resources on the cluster (in reality this would be a superset of the ones
+				// contained in the apisnapshot above, but we use non-overlapping resource names
+				// in this test just to show that we are getting the ones from k8s instead of the
+				// snapshot)
```
Comment (jbohanon) [inline projects/gloo/pkg/servers/iosnapshot/history_test.go:517]: This explanation is so clear and helpful :tada:

**Review by jbohanon — APPROVED** [review]
This is awesome! :rocket:

I think there is some opportunity for serving a specific resource type on `/snapshots/input/{kind}` or `/snapshots/input/{groupkind}` but even just with jq magic, this is so useful e.g.
```sh
curl -s localhost:9091/snapshots/input | jq '.[] | select(.kind=="GatewayParameters" or .kind=="RouteOption" or .kind=="Settings") | .kind, .spec, .status'
```

**Review by sam-heilbron — APPROVED** [review]

File: projects/gloo/pkg/servers/iosnapshot/history.go (hunk 1 above)
Comment (jenshu) [reply projects/gloo/pkg/servers/iosnapshot/history.go:114]: if settings was nil, gloo would have crashed before it got here, but yeah we should be defensive and check for it.
i'll handle this in a follow-up where we improve the way we handle/return errors in the admin server endpoints

**Pull request merged**